src/MatrixClientPeg.js

```diff
@@ -193,14 +193,21 @@ class MatrixClientPeg {
      * Throws an error if unable to deduce the homeserver name
      * (eg. if the user is not logged in)
      */
-    getHomeServerName() {
+    getHomeserverName() {
         const matches = /^@.+:(.+)$/.exec(this.matrixClient.credentials.userId);
         if (matches === null || matches.length < 1) {
             throw new Error("Failed to derive homeserver name from user ID!");
         }
         return matches[1];
     }
 
+    /**
+     * @deprecated Use getHomeserverName() instead (correct spelling)
+     */
+    getHomeServerName() {
```

Comment:
Having two differently-spelled methods seems like a big eyesore. Can't we just replace all usages for  getHomeServerName with getHomeserverName?

Reply:
This PR does that, but by nature of us being an SDK I don't want to break downstream projects.

Reply:
But we appear to regularly make changes like removing props, changing the meaning of params, etc. that would all be breaking changes for downstream... Personally, I find it quite unclear whether we claim to make an compatibility guarantees with the React SDK at all...

Reply:
decision from the standup: sorry everyone who uses the react-sdk, this is the least of your concerns.